ui/ducks/send/send.js

```diff
@@ -0,0 +1,1325 @@
+    // provided payload is always used. For MAX it is calculated based on avail
+    // asset balance
+    mode: AMOUNT_MODES.INPUT,
+    // Current value of the transaction, how much of the asset are we sending
+    value: '0x0',
+    // error to display for amount field
+    error: null,
+  },
+  asset: {
+    // type can be either NATIVE such as ETH or TOKEN for ERC20 tokens
+    type: ASSET_TYPES.NATIVE,
+    // the balance the user holds at the from address for this asset
+    balance: '0x0',
+    // In the case of tokens, the address, decimals and symbol of the token
+    // will be included in details
+    details: null,
+  },
+  draftTransaction: {
+    // The metamask internal id of the transaction. Only populated in theEDIT
+    // stage.
+    id: null,
+    // The hex encoded data provided by the user who has enabled hex data field
+    // in advanced settings
+    userInputHexData: null,
+    // The txParams that should be submitted to the network once this
+    // transaction is confirmed. This object is computed on every write to the
+    // slice of fields that would result in the txParams changing
+    txParams: {
+      to: '',
+      from: '',
+      data: undefined,
+      value: '0x0',
+      gas: '0x0',
+      gasPrice: '0x0',
+    },
+  },
+  recipient: {
+    // Defines which mode to use for searching for matches in the input field
+    mode: RECIPIENT_SEARCH_MODES.CONTACT_LIST,
+    // Partial, not yet validated, entry into the address field. Used to share
+    // user input amongst the AddRecipient and EnsInput components.
+    userInput: '',
+    // The address of the recipient
+    address: '',
+    // The nickname stored in the user's address book for the recipient address
+    nickname: '',
+    // Error to display on the address field
+    error: null,
+  },
+};
+
+const slice = createSlice({
+  name,
+  initialState,
+  reducers: {
+    /**
+     * update current amount.value in state and run post update validation of
+     * the amount field and the send state. Recomputes the draftTransaction
+     */
+    updateSendAmount: (state, action) => {
+      state.amount.value = addHexPrefix(action.payload);
+      // Once amount has changed, validate the field
+      slice.caseReducers.validateAmountField(state);
+      // validate send state
+      slice.caseReducers.validateSendState(state);
+      // Recompute the draftTransaction object
+      slice.caseReducers.updateDraftTransaction(state);
+    },
+    /**
+     * computes the maximum amount of asset that can be sent and then calls
+     * the updateSendAmount action above with the computed value, which will
+     * revalidate the field and form and recomputes the draftTransaction
+     */
+    updateAmountToMax: (state) => {
+      let amount = '0x0';
+      if (state.asset.type === ASSET_TYPES.TOKEN) {
+        const decimals = state.asset.details?.decimals ?? 0;
+        const multiplier = Math.pow(10, Number(decimals));
```

Comment:
We should probably use BN here, b/c a number with enough decimals could exceed JS number precision.
uint256 => 2 ^256 => 1.157920892373162e+77 => as many as 77 decimals

Reply:
I would like to handle this in a new PR because this code wasn't created in this PR and I want to minimize risk of this PR changing current behavior.